clients/src/main/java/org/apache/kafka/common/requests/ProduceResponse.java

```diff
@@ -223,6 +268,8 @@ protected Struct toStruct(short version) {
                 if (partStruct.hasField(LOG_APPEND_TIME_KEY_NAME))
                     partStruct.set(LOG_APPEND_TIME_KEY_NAME, part.logAppendTime);
                 partStruct.setIfExists(LOG_START_OFFSET_FIELD, part.logStartOffset);
+                partStruct.setIfExists(ERROR_RECORDS_KEY_NAME, part.errorRecords.entrySet().toArray());
```

Comment:
This looks broken. We need to have an array of Object, where each object is an instance of Struct. We should write a test case in `RequestResponseTest` which verifies this serialization.